### PR TITLE
Add relogin on connect

### PR DESCRIPTION
**Pull request by ns1123**

### CODE REVIEW

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 1]
```diff
@@ -199,6 +199,7 @@ public synchronized void connect() throws AlluxioStatusException {
     disconnect();
     Preconditions.checkState(!mClosed, "Client is closed, will not try to connect.");
 
+    mContext.getUserState().relogin();
```
Comment (ggezer) [inline core/common/src/main/java/alluxio/AbstractClient.java]: Can you put this to `beforeConnect()` ?

Comment (ns1123) [reply core/common/src/main/java/alluxio/AbstractClient.java]: beforeConnect() would be too late. The relogin() must happen before getAddress() which is called in the try block before the beforeConnect().

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by ns1123** [comment]
alluxio-bot, cherry-pick this to branch-2.1

**Comment by ns1123** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Comment by ns1123** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Pull request merged**